compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/lower/JvmLateinitLowering.kt

```diff
@@ -99,59 +97,28 @@ class JvmLateinitLowering(
             if (irValue !is IrVariable || !irValue.isLateinit) {
                 return expression
             }
-            val irBackingVar = backingVariables[irValue]
-                ?: throw AssertionError("Lateinit variable reference before use: ${expression.dump()}")
 
             return backendContext.createIrBuilder(
-                (irBackingVar.parent as IrSymbolOwner).symbol,
+                (irValue.parent as IrSymbolOwner).symbol,
                 expression.startOffset,
                 expression.endOffset
             ).run {
                 irIfThenElse(
                     expression.type,
-                    irEqualsNull(irGet(irBackingVar)),
-                    backendContext.throwUninitializedPropertyAccessException(this, irBackingVar.name.asString()),
-                    irGet(irBackingVar)
+                    irEqualsNull(irGet(irValue)),
+                    backendContext.throwUninitializedPropertyAccessException(this, irValue.name.asString()),
+                    irGet(irValue)
                 )
             }
         }
 
-        override fun visitSetValue(expression: IrSetValue): IrExpression {
-            expression.transformChildrenVoid(this)
-            val irValue = expression.symbol.owner
-            if (irValue !is IrVariable || !irValue.isLateinit) {
-                return expression
-            }
-
-            val irBackingVar = backingVariables[expression.symbol.owner]
-                ?: throw AssertionError("Lateinit variable reference before use: ${expression.dump()}")
-            return with(expression) {
-                IrSetValueImpl(startOffset, endOffset, type, irBackingVar.symbol, value, origin)
-            }
-        }
-
         override fun visitGetField(expression: IrGetField): IrExpression {
             expression.transformChildrenVoid(this)
             val irField = expression.symbol.owner
-            if (!irField.isLateinitBackingField()) {
-                return expression
-            }
-            val newField = getOrBuildLateinitBackingField(irField)
-            return with(expression) {
-                IrGetFieldImpl(startOffset, endOffset, newField.symbol, newField.type, receiver, origin, superQualifierSymbol)
-            }
-        }
-
-        override fun visitSetField(expression: IrSetField): IrExpression {
-            expression.transformChildrenVoid(this)
-            val irField = expression.symbol.owner
-            if (!irField.isLateinitBackingField()) {
-                return expression
-            }
-            val newField = getOrBuildLateinitBackingField(irField)
-            return with(expression) {
-                IrSetFieldImpl(startOffset, endOffset, newField.symbol, receiver, value, type, origin, superQualifierSymbol)
+            if (irField.isLateinitBackingField()) {
+                irField.type = irField.type.makeNullable()
```

Comment:
I guess we need to change the type of `expression` here, not `irField`

Reply:
Yes, but field too

Reply:
The type of the field will be changed in `visitVariable` though, won't it?

Reply:
`visitVariable` no, but it will be changed in `visitField`, so yes no need to change it here